github/messages.go

```diff
@@ -117,3 +120,52 @@ func validateSignature(signature string, payload, secretKey []byte) error {
 	}
 	return nil
 }
+
+// getWebhookEvent parses the payload and request, and build the Event.
+func getWebhookEvent(r *http.Request, payload []byte) (interface{}, error) {
+	eventType := r.Header.Get(eventTypeHeader)
+	if eventType == "push" {
+		pushEvent := WebHookPayload{}
+		if err := json.Unmarshal(payload, &pushEvent); err != nil {
+			return nil, err
+		}
+		return pushEvent, nil
+	}
+
+	event := Event{
+		Type:       &eventType,
+		RawPayload: (*json.RawMessage)(&payload),
+	}
+	return event.Payload(), nil
```

Comment:
Yes, you're right. Unfortunately they are not the same type strings. Still, I wish go-github would include a piece of code that makes it easier to receive/parse webhooks.

Would it make sense to include a nice way to do it in go-github? Upcast made by the library, client code called with the actual object.